src/pages/resultsView/survival/SurvivalChart.tsx

```diff
@@ -0,0 +1,345 @@
+    }
+
+    @computed get sortedUnalteredPatientSurvivals(): PatientSurvival[] {
+        return this.props.unalteredPatientSurvivals.sort((a, b) => a.months - b.months);
+    }
+
+    @computed get alteredEstimates(): number[] {
+        return getEstimates(this.sortedAlteredPatientSurvivals);
+    }
+
+    @computed get unalteredEstimates(): number[] {
+        return getEstimates(this.sortedUnalteredPatientSurvivals);
+    }
+
+    @computed get logRank(): number {
+        return calculateLogRank(this.sortedAlteredPatientSurvivals, this.sortedUnalteredPatientSurvivals);
+    }
+
+    private tooltipMouseEnter(): void {
+        this.isTooltipHovered = true;
+    }
+
+    private tooltipMouseLeave(): void {
+        this.isTooltipHovered = false;
+        this.tooltipModel = null;
+    }
+
+    private downloadSvg() {
+        this.svgsaver.asSvg(this.svgContainer.firstChild, this.props.fileName + '.svg');
+    }
+
+    private downloadPng() {
+        this.svgsaver.asPng(this.svgContainer.firstChild, this.props.fileName + '.png');
+    }
+
+    public render() {
+
+        const events = [{
+            target: "data",
+            eventHandlers: {
+              onMouseOver: () => {
+                return [
+                  {
+                    target: "data",
+                    mutation: () => ({ active: true })
+                  },
+                  {
+                    target: "labels",
+                    mutation: (props:any) => { 
+                        this.tooltipModel = props;
+                        this.tooltipCounter++;
+                    }
+                  }
+                ];
+              },
+              onMouseOut: () => {
+                return [
+                  {
+                    target: "data",
+                    mutation: () => ({ active: false })
+                  },
+                  {
+                    target: "labels",
+                    mutation: async () => {
+                        await sleep(100);
+                        if (!this.isTooltipHovered && this.tooltipCounter === 1) {
+                            this.tooltipModel = null;
+                        }
+                        this.tooltipCounter--;
+                    }
+                  }
+                ];
+              }
+            }
+          }];
+
+        return (
+            
+            <div className={styles.SurvivalChart}>
+                <VictoryChart containerComponent={<VictoryContainer responsive={false} containerRef={(ref:any) => this.svgContainer = ref}/>}
+                height={650} width={1150} padding={{top: 50, bottom: 50, left: 60, right: 300}}>
+                    <VictoryLabel x={50} y={15} text={this.props.title} style={{fontSize: 24}}/>
+                    <VictoryLine interpolation="stepAfter" data={getLineData(this.sortedAlteredPatientSurvivals, this.alteredEstimates)}
+                    style={{data: {stroke: "red", strokeWidth: 1}}}/>
```

Comment:
According to our current lint settings, this is the preferred indentation for multi-line JSX components (one property per line):

```JavaScript
<VictoryLine 
    interpolation="stepAfter" 
    data={getLineData(this.sortedAlteredPatientSurvivals, this.alteredEstimates)}
    style={{data: {stroke: "red", strokeWidth: 1}}}
/>
```

Reply:
I tried to do this but it gets too long. Maybe we should change the rule :)